### PR TITLE
Fix multiple issues importing Keras models

### PR DESCRIPTION
**Pull request by eusdougc**
This pull request resolves issues #117 and #49.  The changes include:

1) New parameters for _from_keras_model_() interface to allow user to scale and standardize input images

2) Correctly handle additional parameters for batch normalization layer (epsilon additive factor)

3) Allow user to specify label length for user-supplied CSV label file

4) Correctly handle HDF5 files with layers that include '/' in the layer names.

5) Return descriptive error if HDF5 weight file does not have the expected structure.

### CODE REVIEW

**Comment by eusdougc** [comment]
Regarding 2, the sample notebook that I am going to commit will address issue #49.  The problem I have with making test cases for anything that involves importing model weights is that the process tends to be slow.  I believe the philosophy for the test cases is that they should run quickly, so I have somewhat of a dilemma.  Perhaps I can make a very simple test case - I will investigate.

**Comment by ethem-kinginthenorth** [comment]
Thanks. yes you are 100% right. unit tests should be quick, but we are already growing our code base a lot so I am happy to bear with slow tests :), I don't want anyone else to break the changes that you have been exclusively making as they are very important to us.

**Comment by eusdougc** [comment]
I added a new test case for Keras model conversion.

File: dlpy/model_conversion/write_sas_code.py [hunk 1]
```diff
@@ -67,27 +85,27 @@ def write_input_layer(model_name='sas', layer_name='data', channels='-1',
...
         '   # input layer',
         '   nchannels=' + channels,
         '   if input_channel_offset is None and nchannels==3:',
-        '       print("INFO: setting channel mean values to ImageNet means")',
+        '       print("INFO: Setting channel mean values to ImageNet means")',
         '       input_channel_offset = [103.939, 116.779, 123.68]',
         '       s.addLayer(model=' + repr(model_name) + ', name=' + repr(layer_name) + ',',
         '                  layer=dict( type="input", nchannels=' + channels + ', width=' + width + ', height=' + height + ',',
-        '                  scale = ' + scale + ', randomcrop=input_crop_type, offsets=input_channel_offset))',
+        '                  scale = ' + scale + ', randomcrop=input_crop_type, offsets=input_channel_offset, offsetStd=norm_std))',
```
Comment (kesmit13) [inline dlpy/model_conversion/write_sas_code.py]: I'm guessing that offsetStd is a parameter to addLayer and that's why it isn't underscore-delimited?

File: dlpy/network.py [hunk 2]
```diff
@@ -810,10 +829,13 @@ def load_weights_from_keras(self, path, labels=False, data_spec=None, label_file
         label_file_name: string, optional
             Fully qualified path to CSV file containing user-defined
             classification labels.  If not specified, ImageNet labels assumed.
+        label_length: int, optional
```
Comment (kesmit13) [inline dlpy/network.py]: Make sure you leave a space before the ':' in docstring lines.  This will mess up Sphinx when parsing.  Make sure to correct others above as well.

File: dlpy/tests/test_model_conversion.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    '''
+    Please locate the images.sashdat file under the datasources to the DLPY_DATA_DIR.
+    '''
+    server_type = None
+    s = None
+    server_sep = '/'
+    data_dir = None
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (kesmit13) [inline dlpy/tests/test_model_conversion.py]: setUpClass should be underscore-delimited

File: dlpy/model_conversion/write_sas_code.py (hunk 1 above)
Comment (eusdougc) [reply dlpy/model_conversion/write_sas_code.py]: That is correct.

File: dlpy/network.py (hunk 2 above)
Comment (eusdougc) [reply dlpy/network.py]: I will fix and re-submit.

**Comment by eusdougc** [comment]
I addressed the issue with ":" in docstring lines and added a check in my new test to make sure it doesn't crash if a file is missing.

**Comment by eusdougc** [comment]
I just added a binary HDF5 file to the datasources subdirectory.  This file is used by my new Keras model conversion test.

**Comment by eusdougc** [comment]
I just committed two minor changes to sas_keras_parse.py to fix a bug I just discovered.
